scripts/finish-release.ts

```diff
@@ -0,0 +1,247 @@
+import * as execa from 'execa';
+import * as rl from 'readline';
+import { stdin, stdout } from 'process';
+import yargs from 'yargs';
+
+export function extractUpstreamNameFromRemotes(remoteGitVerbose: string, upstreamRepositoryName: string): string | undefined {
+  const lineWithRepoName = remoteGitVerbose.split('\n').find((l) => l.indexOf(upstreamRepositoryName) > -1);
+  return lineWithRepoName?.split(/\s/)?.[0];
+}
+
+export interface MergeOptions {
+  message: string;
+  mode: 'ff-only';
+}
+export class Git {
+  remote(verbose?: boolean): string {
+    let command = ['git', 'remote'];
+    if (verbose) {
+      command.push('-v');
+    }
+    return execa.sync(command[0], command.slice(1)).stdout;
```

Comment:
nit: since it looks like all of these methods execute git commands, could we just define 'git' as a private readonly string and reuse everywhere rather than all this array indexing?

Reply:
I have a follow on PR queued up that will extract the git module to its own file. I'll address this in that change.